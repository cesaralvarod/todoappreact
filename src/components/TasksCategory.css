.tasks-category {
  margin-bottom: 3rem;
}

.tasks-category h3 {
  border-bottom: 1px solid #eee;
  padding-bottom: 2rem;
}

.tasks-category p.total-p {
  margin-top: 2rem;
  font-size: 1.4rem;
  text-align: right;
}
