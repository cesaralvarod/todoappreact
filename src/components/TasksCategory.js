import React from "react";
import PropTypes from "prop-types";

// Components

import TasksList from "./TasksList";

// Styles

import "./TasksCategory.css";

const TasksCategory = ({ title, tasks }) => {
  return (
    <div className="tasks-category">
      {title && <h3>{title}</h3>}
      <TasksList tasks={tasks} />
      <p className="total-p">Total: {tasks.length}</p>
    </div>
  );
};

TasksList.propTypes = {
  title: PropTypes.string,
  tasks: PropTypes.array,
};

export default TasksCategory;
