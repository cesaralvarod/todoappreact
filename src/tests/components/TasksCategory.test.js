import React from "react";
import { shallow } from "enzyme";
import TasksCategory from "../../components/TasksCategory";
import { demoData } from "../fixtures/demoData";

describe("Testing in component TasksCategory", () => {
  const wrapper = shallow(
    <TasksCategory title="Category ..." tasks={demoData.data} />
  );

  test("Component TaskCategory should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
