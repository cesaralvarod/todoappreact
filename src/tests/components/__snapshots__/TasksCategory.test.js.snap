// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component TasksCategory Component TaskCategory should show correctly 1`] = `
<div
  className="tasks-category"
>
  <h3>
    Category ...
  </h3>
  <TasksList
    tasks={
      Array [
        Object {
          "content": "Learn react",
          "done": false,
          "id": 0,
        },
        Object {
          "content": "Do homework",
          "done": true,
          "id": 1,
        },
        Object {
          "content": "Watch tv",
          "done": false,
          "id": 2,
        },
      ]
    }
  />
  <p
    className="total-p"
  >
    Total: 
    3
  </p>
</div>
`;
